recipes/graphvite/build-pkg.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -ex
+
+mkdir -p build
+cd build
+
+# function for facilitate version comparison; cf. https://stackoverflow.com/a/37939589
+function version2int { echo "$@" | awk -F. '{ printf("%d%02d\n", $1, $2); }'; }
+
+# for documentation see e.g.
+# docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#building-for-maximum-compatibility
+# docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#ptxas-options-gpu-name
+# docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#gpu-feature-list
+
+# the following are all the x86-relevant gpu arches
+ARCHES=(52 60 61 70)
+if [ $(version2int $cuda_compiler_version) -ge $(version2int "11.1") ]; then
+    # Ampere support for GeForce 30 (sm_86) needs cuda >= 11.1
+    LATEST_ARCH=86
+    # ARCHES does not contain LATEST_ARCH; see usage below
+    ARCHES=( "${ARCHES[@]}" 75 80 )
+elif [ $(version2int $cuda_compiler_version) -ge $(version2int "11.0") ]; then
+    # Ampere support for A100 (sm_80) needs cuda >= 11.0
+    LATEST_ARCH=80
+    ARCHES=( "${ARCHES[@]}" 75 )
+fi
+for arch in "${ARCHES[@]}"; do
+    CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS+${CMAKE_CUDA_ARCHS};}${arch}-real"
+done
+# for -real vs. -virtual, see cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html
+# this is to support PTX JIT compilation; see first link above or cf.
+# devblogs.nvidia.com/cuda-pro-tip-understand-fat-binaries-jit-caching
+CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS+${CMAKE_CUDA_ARCHS};}${LATEST_ARCH}"
+
+cmake \
+    ${CMAKE_ARGS} \
+    -DPROJECT_BINARY_DIR=$PREFIX \
+    -DFAISS_PATH=$PREFIX/lib/libfaiss.so \
+    -DCMAKE_CUDA_ARCHITECTURES="${CMAKE_CUDA_ARCHS}" \
+    ..
+
+make
+make install
+
+cd ../python
+$PYTHON setup.py install
```

Comment:
@h-vetinari it should be $PREFIX in the patch. You call setup during the build and it should look for $PREFIX on line 38 above. I faced a similar problem packaging mxnet 😢

Reply:
See https://github.com/conda-forge/staged-recipes/pull/16943/commits/f5bc2420dc645d0a42aabb04a070296a0e09f817

Reply:
Yeah, I saw. I just wouldn't add the CONDA_PREFIX 😄 

Good job btw!! This is exciting

Reply:
> Yeah, I saw. I just wouldn't add the CONDA_PREFIX 😄

If you open a locally installed environment, `$PREFIX` isn't going to be populated, but `$CONDA_PREFIX` is. Hence we need to reflect that.

Reply:
Why would that be relevant? Does this recipe look for other libraries besides libgraphvite?  `lib_file = imp.find_module("libgraphvite", candidate_paths)[1]`

setup.py will never be triggered after the build stage here afaik

Reply:
This is patching `python/graphvite/__init__.py`...

Reply:
Oopoppopps... I thought it was setup.py